lib/plug/request_id.ex

```diff
@@ -56,26 +63,21 @@ defmodule Plug.RequestId do
 
   @impl true
   def call(conn, {header, assign_as}) do
-    conn
-    |> get_request_id(header)
-    |> set_request_id(header, assign_as)
-  end
-
-  defp get_request_id(conn, header) do
-    case Conn.get_req_header(conn, header) do
-      [] -> {conn, generate_request_id()}
-      [val | _] -> if valid_request_id?(val), do: {conn, val}, else: {conn, generate_request_id()}
-    end
-  end
+    request_id = get_request_id(conn, header)
```

Comment:
We don't need to pass in the `conn` to `get_request_id/2` and return it, because we don't update it. One less tuple allocation and one less function call.